crates/ide-completion/src/render/function.rs

```diff
@@ -79,6 +79,11 @@ fn render(
         .and_then(|trait_| trait_.containing_trait_or_trait_impl(ctx.db()))
         .map_or(false, |trait_| completion.is_ops_trait(trait_));
 
+    let is_item_from_notable_trait = func
+        .as_assoc_item(ctx.db())
```

Comment:
Let's bind the `as_assoc_item` result to var and re-use that as multiple things in this function are interested in it (and the call is not completely free as it accesses the database)

Reply:
Is it okay, if bind the result of the `and_then` as well, since seems like that's call to the db as well? I assume both methods (`containing_trait` and `containing_trait_or_trait_impl`) return the needed `Trait`.

Also is `ctx.db()` same as the `db` in the parameter of the render function?

Reply:
The `db` is the same ye. You can (and actually want) to use `containing_trait_or_trait_impl` for both here, I missed that actually (since the attribuite is only on the trait, not trait impls).

Reply:
I thought the same too but just now saw the implementation and figured to ask the question.